src/cljc/rems/common/catalogue_util.cljc

```diff
@@ -18,6 +18,16 @@
   (when (and enable-ega (ega-dataset? resid))
     (str (or ega-organization "https://ega-archive.org/datasets/") resid)))
 
+;; https://www.crossref.org/blog/dois-and-matching-regular-expressions/
+;; "For the 74.9M DOIs we have seen this matches 74.4M of them. If you need to use only one pattern then use this one."
+(defn- doi? [resid]
+  (and resid (re-find #"(?i)10.\d{4,9}[-._;()/:A-Z0-9]+$" resid)))
+
+(defn- doi-catalogue-item-url [resid {:keys [enable-doi doi-organization]}]
+  (when enable-doi
+    (some->> (doi? resid)
+             (str (or doi-organization "https://doi.org/")))))
```

Comment:
I think this is slightly clever code and the simple stupid version would be:
```clj
(when enable-doi
  (when-let [doi (doi? resid)]
    (str (or doi-organization "https://doi.org/") doi))
```

Reply:
For some functions, we have had a convention that a function with `?` will only return a boolean. I'm not a strong proponent of that, I am fine with the typical "truthy" but we could name `doi?` perhaps like `parse-doi` to clarify this.

Reply:
good points!

1. i stumbled upon `some->>` previously and it seemed pretty cool, but there is a bit of extra cognitive load over nested `when`:s
2. true, this function does return either `nil` or the parsed string, so it's not a good candidate for `?`